File: Base/src/main/java/com/example/board/global/error/exception/ErrorCode.java
package com.example.board.global.error.exception;

import lombok.AllArgsConstructor;
import lombok.Getter;

@Getter
@AllArgsConstructor
public enum ErrorCode {

    EXPIRED_TOKEN(401, "Expired Token"),
    INVALID_TOKEN(401, "Invalid Token"),
    PASSWORD_MISMATCHED(401, "Password Mismatched"),

    USER_NOT_FOUND(404, "User Not Found"),
    REFRESH_TOKEN_NOT_FOUND(404, " Refresh Token Not Found"),

    ALREADY_USER_EIXTS(409, "Already User Exits"),

    INTERNAL_SERVER_ERROR(500, "Internal Server Error");

    private final int status;
    private final String message;
}

File: Kopring/Dise_V2/src/main/kotlin/com/example/dise_v2/global/error/ErrorCode.kt
package com.example.dise_v2.global.error

import org.springframework.http.HttpStatus

enum class ErrorCode(
    val status: Int,
    val message: String
) {
    EXPIRED_JWT(401, "만료된 토큰입니다."),
    INVALID_JWT(401, "유효하지 않은 토큰입니다."),
    PASSWORD_MISMATCH(401, "비밀번호가 불일치합니다."),
    NOT_MODIFY_FEED(401, "게시글을 수정할 수 없습니다."),

    USER_NOT_FOUND(404, "유저를 찾을 수 없습니다."),
    FEED_NOT_FOUND(404, "게시물을 찾을 수 없습니다."),

    ALREADY_USER_EXIST(409, "이미 존재하는 유저입니다."),

    INTERNAL_SERVER_ERROR(500, "내부 서버 오류입니다.");
}